ext/js/language/language-transforms.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2024  Yomitan Authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+
+/**
+ * @param {string} inflectedSuffix
+ * @param {string} deinflectedSuffix
+ * @param {string[]} conditionsIn
+ * @param {string[]} conditionsOut
+ * @returns {import('language-transformer').SuffixRule}
+ */
+export function suffixInflection(inflectedSuffix, deinflectedSuffix, conditionsIn, conditionsOut) {
+    const suffixRegExp = new RegExp(inflectedSuffix + '$');
+    return {
+        type: 'suffix',
+        isInflected: suffixRegExp,
+        deinflected: deinflectedSuffix,
+        uninflect: (text) => text.replace(suffixRegExp, deinflectedSuffix),
+        conditionsIn,
+        conditionsOut
+    };
+}
+
+/**
+ * @param {string} inflectedPrefix
+ * @param {string} deinflectedPrefix
+ * @param {string[]} conditionsIn
+ * @param {string[]} conditionsOut
+ * @returns {import('language-transformer').Rule}
+ */
+export function prefixInflection(inflectedPrefix, deinflectedPrefix, conditionsIn, conditionsOut) {
+    const prefixRegExp = new RegExp('^' + inflectedPrefix);
+    return {
+        type: 'prefix',
+        isInflected: prefixRegExp,
+        uninflect: (text) => text.replace(prefixRegExp, deinflectedPrefix),
```

Comment:
Another thought on the use of regex, is this actually more efficient compared to using `string.endsWith` and then just doing a substring? At a minimum, it would seem to be using more memory to create the RegExp and store it in a closure.

Reply:
I went with replace over endsWith here just because it benched slightly faster, but now I see that closures may have some risks so I think it's worth using the string methods.

Reply:
As for isInflected, maybe we could switch on type and do regex only on rules of type `other`? Though maybe we should just keep it uniform unless there's a performance issue.